.github/workflows/build_and_test.yaml

```diff
@@ -49,6 +49,9 @@ jobs:
           # Only rebuild cache if build.gradle is changed
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: ${{ runner.os }}-gradle
+      - name: Verify Dependencies
+        run: |
+          eval ./gradlew assemble
```

Comment:
@aplex This might be a design issue with how I did the pipeline, but I also added some caching on the build step to shorten build times. I think it ends up caching dependencies so transitive dependency issues don't get detected by the pipeline.

Reply:
I see that we skip test on the normal build below. I think that will also skip compilation and dependency resolution for test code. We can tell gradle to compile tests but not run them yet like this - https://stackoverflow.com/questions/43638697/gradle-compile-tests-but-do-not-run-them 

Did we see those dependency problems in test or in a main code?

Reply:
The dependency problems exist in the main code, I think the tests will compile in the testing step so it should still be caught by the pipeline eventually.